sdk/src/azure/core/az_json_reader.c

```diff
@@ -271,7 +271,8 @@ AZ_NODISCARD static az_result _az_json_reader_process_string(az_json_reader* ref
     {
       break;
     }
-    else if (next_byte == '\\')
+
```

Comment:
Was changing the `else if` to `if` intended?  Is there a test case for this bug?

Reply:
Yes, it was intentional. Yes, we have test coverage. No, it isn't a bug.

This is in no way changing the execution/behavior of the code. It is merely a readability fix since we don't need an `else if` following a `break`, similar to how we don't need `else if` after an `if` block that `returns`.

It's this check: `readability-else-after-return`